tests/splines/batched_2d_spline_builder.cpp

```diff
@@ -0,0 +1,760 @@
+template <class IDimI1, class IDimI2, class T>
+struct DimsInitializer;
+
+template <class IDimI1, class IDimI2, class... IDimX>
+struct DimsInitializer<IDimI1, IDimI2, ddc::detail::TypeSeq<IDimX...>>
+{
+    void operator()(std::size_t const ncells)
+    {
+#if defined(BSPLINES_TYPE_UNIFORM)
+        (ddc::init_discrete_space(IDimX::
+                                          init(x0<typename IDimX::continuous_dimension_type>(),
+                                               xN<typename IDimX::continuous_dimension_type>(),
+                                               DVect<IDimX>(ncells))),
+         ...);
+        ddc::init_discrete_space<BSplines<typename IDimI1::continuous_dimension_type>>(
+                x0<typename IDimI1::continuous_dimension_type>(),
+                xN<typename IDimI1::continuous_dimension_type>(),
+                ncells);
+        ddc::init_discrete_space<BSplines<typename IDimI2::continuous_dimension_type>>(
+                x0<typename IDimI2::continuous_dimension_type>(),
+                xN<typename IDimI2::continuous_dimension_type>(),
+                ncells);
+#elif defined(BSPLINES_TYPE_NON_UNIFORM)
+        (ddc::init_discrete_space<IDimX>(breaks<typename IDimX::continuous_dimension_type>(ncells)),
+         ...);
+        ddc::init_discrete_space<BSplines<typename IDimI1::continuous_dimension_type>>(
+                breaks<typename IDimI1::continuous_dimension_type>(ncells));
+        ddc::init_discrete_space<BSplines<typename IDimI2::continuous_dimension_type>>(
+                breaks<typename IDimI2::continuous_dimension_type>(ncells));
+#endif
+        ddc::init_discrete_space<IDimI1>(
+                GrevillePoints<
+                        BSplines<typename IDimI1::continuous_dimension_type>>::get_sampling());
+        ddc::init_discrete_space<IDimI2>(
+                GrevillePoints<
+                        BSplines<typename IDimI2::continuous_dimension_type>>::get_sampling());
+    }
+};
+
+// Checks that when evaluating the spline at interpolation points one
+// recovers values that were used to build the spline
+template <typename ExecSpace, typename MemorySpace, typename I1, typename I2, typename... X>
+static void Batched2dSplineTest()
+{
+    // Instantiate execution spaces and initialize spaces
+    Kokkos::DefaultHostExecutionSpace host_exec_space = Kokkos::DefaultHostExecutionSpace();
+    ExecSpace exec_space = ExecSpace();
+    std::size_t constexpr ncells = 10;
+    DimsInitializer<
+            IDim<I1, I1, I2>,
+            IDim<I2, I1, I2>,
+            BatchDims<IDim<I1, I1, I2>, IDim<I2, I1, I2>, IDim<X, I1, I2>...>>
+            dims_initializer;
+    dims_initializer(ncells);
+
+    // Create the values domain (mesh)
+    auto interpolation_domain1
+            = ddc::DiscreteDomain<IDim<I1, I1, I2>>(GrevillePoints<BSplines<I1>>::get_domain());
+    auto interpolation_domain2
+            = ddc::DiscreteDomain<IDim<I2, I1, I2>>(GrevillePoints<BSplines<I2>>::get_domain());
```

Comment:
Can you move these 2 definitions into a `#if defined(BC_HERMITE)` block to avoid ` unused variable` warnings ?